### PR TITLE
The last moment for moment

### PR DESCRIPTION
**Pull request by GalenWong**
- Mandatory moment of silence for moment.js before reviewing this PR: https://www.youtube.com/watch?v=Fw3SHWroQdw
- I chose to replace moment.js with day.js since the set of API we use in the code base has the same format in dayjs. 
- A comparison for the functionality is found here: https://github.com/you-dont-need/You-Dont-Need-Momentjs#feature-parity
- we can't simply change the import from `import moment from 'moment'` to `import dayjs from 'dayjs'`. Some APIs require extra plugins since day.js opted for keeping the package as lightweight as possible. The list of functions that require plugin is found in the comparison document above and in the day.js docs https://day.js.org/docs/en/plugin/plugin

### CODE REVIEW

**Comment by GalenWong** [comment]
You might wonder why we still have moment in `yarn.lock. Here's why:

```
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "moment"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "moment@2.29.1"
info Reasons this module exists
   - "gatsby" depends on it
   - Hoisted from "gatsby#moment"
info Disk size without dependencies: "5.14MB"
info Disk size with unique dependencies: "5.14MB"
info Disk size with transitive dependencies: "5.14MB"
info Number of shared dependencies: 0
✨  Done in 1.65s.
```

**Review by TimothyGu — APPROVED** [review]

File: gatsby-node.js [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 const path = require('path');
 const { createFilePath } = require('gatsby-source-filesystem');
-const moment = require('moment');
 
 const events = require('./src/data/events/events');
 const highlightedEvents = require('./src/data/events/highlights');
+const dayjs = require('dayjs');
```
Comment (TimothyGu) [inline gatsby-node.js:12]:
```suggestion
const dayjs = require('dayjs');

const events = require('./src/data/events/events');
const highlightedEvents = require('./src/data/events/highlights');
```

Comment (GalenWong) [reply gatsby-node.js:12]: pay attention in class man

Comment (TimothyGu) [reply gatsby-node.js:12]: y tú, sir

**Pull request merged**